llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -6008,6 +6009,15 @@ static bool combineInstructionsOverFunction(
     if (!MadeChangeInThisIteration)
       break;
 
+    // Issue #147390: InstCombine emits `select` as rewrites of instructions
+    // without metadata. It may be possible to synthesize some more meaningful
+    // profiles in some cases (based on operands or based on the pattern)
+    if (auto EC = F.getEntryCount(); EC.has_value() && EC->getCount() > 0)
+      for (auto &BB : F)
+        for (auto &I : BB)
+          if (isa<SelectInst>(I) && !I.getMetadata(LLVMContext::MD_prof))
```

Comment:
If we have a select instruction with no profile metadata before `instcombine`, this would also mark the branch weights as explicitly unknown.

Is that something that we want?

Reply:
Nitpicking: the profcheck verifier doesn't run after every pass, i.e. it doesn't affect compile time. It runs around every .ll test (on the profcheck bot). Meaning, it's possible selects would arrive here without `!prof` _if_ they get inserted in a pass with that behavior left untested.